image/src/utils/csv.js

```diff
@@ -0,0 +1,80 @@
+const fetch = require('node-fetch')
+const { FILENAME, DELIMITER, LIMIT_TYPE, LIMIT_SIZE, EGRESS_URL } = require('../config/config')
+const fs = require('fs')
+const { isValidURL } = require('./util')
+
+const processPayload = async json => {
+  const data = json.data !== undefined ? json.data : json
+  if (fs.existsSync(FILENAME)) {
+    const stats = fs.statSync(FILENAME)
+    const size = stats.size
+    const fileData = fs.readFileSync(FILENAME)
+    const lines = fileData.toString().split('\n').length
+    if ((LIMIT_TYPE === 'size' && size > LIMIT_SIZE) || (LIMIT_TYPE === 'rows' && lines > LIMIT_SIZE)) await upload()
+    await writeData(data, true)
```

Comment:
So you upload the file only when the size is greater than the limit size. I think that the limit is there so the file is not greater. I think the file should be sent when file size is just below the limit and no more data can be fit in. Maybe before you add new data to the file you check if you don't exceed the limit. If you don't then you add data, if you exceed then you upload the file and then add the data.

The second thing is that you upload the file and then you call `writeData(data, true)` so you append data to the same file? So you keep increasing the size of the file above the limit size? Why don't you clear the file once you upload it and the keep saving new data to it?

Reply:
the file is uploaded when it exceeds the limit, it's checked on every new request. then file is uploaded, deleted and new (incoming) data is written in same newly created file. unless for some reason file upload fails, then it's not deleted, to prevent data loss, it will be uploaded again in next request. so that data is not lost.